radixdlt/src/integration/java/com/radixdlt/consensus/PerfectNetworkTest.java

```diff
@@ -65,11 +69,22 @@ public void given_3_correct_bfts__then_all_should_get_same_commits_and_no_timeou
 		Observable<Object> timeoutCheck = Observable.interval(2, TimeUnit.SECONDS)
 			.flatMapIterable(i -> nodes)
 			.map(bftNetwork::getCounters)
-			.doOnNext(counters -> assertThat(counters.getCount(CounterType.TIMEOUT)).isEqualTo(0))
+			.doOnNext(counters -> assertThat(counters.getCount(CounterType.TIMEOUT))
+				.satisfies(new Condition<>(c -> c == 0, "Timeout counter is zero.")))
 			.map(o -> o);
 
-		Observable.merge(commitCheck, timeoutCheck)
-			.doOnSubscribe(d -> bftNetwork.start())
+		List<Observable<Vertex>> proposals = nodes.stream()
+			.map(ECKeyPair::getUID)
+			.map(bftNetwork.getTestEventCoordinatorNetwork()::getNetworkRx)
+			.map(EventCoordinatorNetworkRx::proposalMessages)
+			.collect(Collectors.toList());
+
+		Observable<Object> proposalsCheck = Observable.merge(proposals)
+			.doOnNext(v -> assertThat(v)
+				.satisfies(new Condition<>(vtx -> vtx.getView().equals(vtx.getParentView().next()), "Vertex has direct parent")))
```

Comment:
We use this so often, a helper method `hasDirectParent` that checks `this.getView().equals(this.getParentView().next())` might make things a little easier to read.